resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -97,4 +97,17 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
             }
         });
     }
+
+    /**
+     * @param anyInt any integer value to covert
+     * @param len max limit of range
+     * @return integer in (0, len] range, using formula {@code (anyInt % len + anyInt) % len}
+     */
+    protected static int toIndex(int anyInt, int len) {
+        int index = anyInt % len;
+        if (index < 0) {
+            index = index + len;
```

Comment:
Your continuity will break from `Integer.MAX_VALUE` to `Integer.MIN_VALUE` (see my latest code), don't jump too fast to conclusions, one divisor might preserve continuity but not the other.

The only way to guarantee continuity is when the length is a power of two, either way to preserve continuity you would have to amortize.

Either way, it will work even with the jump, but it will jump...I'm not trying to have an argument here, I was just trying to simply understand, but I will stop now for good.